### PR TITLE
[Validator] Add attributes documentation of composite constraints

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Related pull-request: https://github.com/symfony/symfony/pull/41994

Not really sure how to precise it only works for attributes in PHP 8.1, not 8.0. If you have a solution, please let me know!

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Thanks for this Alexander!

I have two comments:

### Nested attribute code syntax

Is there a commonly used syntax for this? I'm asking because instead of this:

```php
    #[
        Assert\All([
            new Assert\NotBlank,
            new Assert\Length(min: 5),
        ])
    ]
    protected $favoriteColors = [];
```

I'd prefer to use this:

```php
    #[Assert\All([
        new Assert\NotBlank,
        new Assert\Length(min: 5),
    ])]
    protected $favoriteColors = [];
```

But this is just a personal preference, so let's ping the @symfony/team-symfony-docs to ask for their opinion.

### Warning message about needing PHP 8.1 to run this code

That's a good question and I don't have a good answer. Maybe we can add a comment just before each class like the following?

```rst
.. code-block:: php-attributes

    // src/Localization/Place.php
    namespace App\Localization;

    use App\Validator\Constraints as AcmeAssert;
    use Symfony\Component\Validator\Constraints as Assert;

    // IMPORTANT: the code of this class requires using PHP 8.1 or higher

    class Place
    {
        // ...
    }
```

But I'm not sure ... so let's ping @symfony/team-symfony-docs again to ask what do they think. Thanks!

File: reference/constraints/AtLeastOneOf.rst [hunk 1]
```diff
@@ -60,6 +60,34 @@ The following constraints ensure that:
...
+
+        class Student
+        {
+            #[
+                Assert\AtLeastOneOf([
+                    new Assert\Regex('/#/'),
+                    new Assert\Length(min: 10),
+                ])
+            ]
+            protected $password;
```
Comment (OskarStark) [inline reference/constraints/AtLeastOneOf.rst]: We should rename this to `plainPassword`, in nearly every application, this should be a hash, which does not need to be validated 🤔

Comment (alexandre-daubois) [reply reference/constraints/AtLeastOneOf.rst]: You're right, making the change!

**Comment by alexandre-daubois** [comment]
Thank you for your feedback @javiereguiluz!

About the code syntax, I have to say that I don't particularly have an opinion either. I based myself on code written by @derrabus, where you can find this in some tests of the framework:

```php
#[
    Assert\NotNull,
    Assert\Range(min: 3),
]
public $firstName;
```

About the warning message for PHP 8.1, for now I'll add the comment like you did and we'll see if a better solution appears.

**Review by derrabus — APPROVED** [review]

**Comment by malteschlueter** [comment]
I think this can be merged now or not?

**Comment by wouterj** [comment]
You're right @malteschlueter.

Thank you @alexandre-daubois! I've slightly updated your changes, to use the CS proposed by Javier and to add versionadded directives below the code examples (to emphasize the fact that these attributes are not supported in 5.3, which is different for all other constraint).

**Pull request merged**